testhelpers/docker_registry.go

```diff
@@ -68,17 +68,24 @@ func NewDockerRegistry(ops ...RegistryOption) *DockerRegistry {
 }
 
 // BasicAuth wraps a handler, allowing requests with matching username and password headers, otherwise rejecting with a 401
-func BasicAuth(handler http.Handler, username, password, realm string) http.Handler {
+// optPermissions is used to skip the authentication check if the image is readable.
+func BasicAuth(handler http.Handler, username, password, realm string, optPermissions ...map[string]ImagePrivileges) http.Handler {
```

Comment:
Why does `optPermissions` need to be passed in if `customPrivileges` is a field on the registry?

Reply:
Because these HTTP handler functions don't have access to those attributes, they are plain functions handling the HTTP request. I think that's why we are also sending the username/password values from the registry.

Reply:
Ah so again, reading more closely, it would be called before `Start()`.

Reply:
Actually, if I am not wrong, the HTTP server is holding the reference to the handler here:
```	
r.server = &httptest.Server{
	Listener: listener,
	Config:   &http.Server{Handler: r.authnHandler},
}
```
every time a new request is coming, the HTTP server is invoking the handler function which is, in fact, referencing the attribute from the docker registry (the permission map), and that is how it is dynamically updated when call the `Add` method  after the `Start`

Reply:
@yaelharel , @natalieparellano.

The `Add` method can actually be called after the `Start`, as I mentioned before, it handler is holding a reference to the internal Map and every time the internal map is being updated after an `Add` invocation when the handler function is called the Map update is reflected.

I added a log entry in the Custom handler function to print the URL of the request and the content of the Map that is received, as you can see, the Map values are updated and the handler is called on every request made to the Docker registry on each test cases.

[local-out.txt](https://github.com/buildpacks/imgutil/files/6970269/local-out.txt)